pkg/deploy/scaler/scale.go

```diff
@@ -63,33 +51,49 @@ func (scaler *DeploymentConfigScaler) Scale(namespace, name string, newSize uint
 		if err != nil {
 			return err
 		}
-		return wait.Poll(waitForReplicas.Interval, waitForReplicas.Timeout, kclient.ControllerHasDesiredReplicas(scaler.clientInterface, rc))
+		return wait.Poll(waitForReplicas.Interval, waitForReplicas.Timeout, controllerHasSpecifiedReplicas(scaler.clientInterface, rc, dc.Template.ControllerTemplate.Replicas))
 	}
 	return nil
 }
 
-// ScaleSimple does a simple one-shot attempt at scaling - not useful on it's own, but
-// a necessary building block for Scale
+// ScaleSimple does a simple one-shot attempt at scaling - not useful on it's
+// own, but a necessary building block for Scale
 func (scaler *DeploymentConfigScaler) ScaleSimple(namespace, name string, preconditions *kubectl.ScalePrecondition, newSize uint) error {
 	dc, err := scaler.dcClient.DeploymentConfigs(namespace).Get(name)
 	if err != nil {
 		return err
 	}
-	controller, err := scaler.rcClient.ReplicationControllers(namespace).Get(util.LatestDeploymentNameForConfig(dc))
-	if err != nil {
-		return kubectl.ControllerScaleError{FailureType: kubectl.ControllerScaleGetFailure, ResourceVersion: "Unknown", ActualError: err}
+	scale := deployapi.ScaleFromConfig(dc)
+	scale.Spec.Replicas = int(newSize)
+	if _, err := scaler.dcClient.DeploymentConfigs(namespace).UpdateScale(scale); err != nil {
+		return kubectl.ControllerScaleError{FailureType: kubectl.ControllerScaleUpdateFailure, ResourceVersion: dc.ResourceVersion, ActualError: err}
 	}
+	// TODO: do a better job of printing objects here.
```

Comment:
remove this TODO, artifact from the time when ScaleSimple and Scale were returning strings